pkg/client/client_test.go

```diff
@@ -57,25 +60,47 @@ func TestApplySingle(t *testing.T) {
 		// given
 		namespacedName := types.NamespacedName{Namespace: "toolchain-host-operator", Name: "registration-service"}
 
-		t.Run("when using forceUpdate=true, it should update even when spec is same", func(t *testing.T) {
+		t.Run("when using forceUpdate=true, it should not update when specs are same", func(t *testing.T) {
 			// given
-			cli := NewFakeClient(t)
-			cl := applyCl.NewApplyClient(cli, s)
-			_, err := cl.CreateOrUpdateObject(defaultService.DeepCopyObject(), true, nil)
+			cl, _ := newClient(t, s)
+			obj := defaultService.DeepCopyObject()
+			_, err := cl.CreateOrUpdateObject(obj, true, nil)
 			require.NoError(t, err)
+			originalGeneration := obj.(*corev1.Service).GetGeneration()
 
-			// when
-			createdOrChanged, err := cl.CreateOrUpdateObject(modifiedService.DeepCopyObject(), true, nil)
+			// when updating with the same obj again
+			createdOrChanged, err := cl.CreateOrUpdateObject(obj, true, nil)
 
 			// then
 			require.NoError(t, err)
-			assert.True(t, createdOrChanged)
+			assert.False(t, createdOrChanged) // resource was not updated on the server, so returned value is `false`
+			updateGeneration := obj.(*corev1.Service).GetGeneration()
+			assert.Equal(t, updateGeneration, originalGeneration)
```

Comment:
it's the other way around - the first parameter should be expected value and the second the actual value:
```suggestion
			assert.Equal(t, originalGeneration, updateGeneration)
```